jlm/llvm/ir/operators/FunctionPointer.hpp

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2024 Helge Bahmann <hcb@chaoticmind.net>
+ * See COPYING for terms of redistribution.
+ */
+
+#ifndef JLM_LLVM_IR_OPERATORS_FUNCTIONPOINTER_HPP
+#define JLM_LLVM_IR_OPERATORS_FUNCTIONPOINTER_HPP
+
+#include <jlm/llvm/ir/tac.hpp>
+#include <jlm/llvm/ir/types.hpp>
+#include <jlm/rvsdg/bitstring/type.hpp>
+#include <jlm/rvsdg/simple-node.hpp>
+
+namespace jlm::llvm
+{
+
+/**
+  \brief Get address of compiled function object.
+  */
+class FunctionToPointerOperation final : public rvsdg::SimpleOperation
```

Comment:
I do not think that the constructor and creators of this class need to take the `PointerType` as argument. It can be explicit I think. The pointer type has no attributes, it is a simply opaque.

Reply:
you mean "implicit"? i.e. remove the pointerType argument and just infer PointerType::Create()?

Reply:
Yes, I meant implicit. My bad. Sorry!

Reply:
okay removed